io.github.fastrizwaan.WineCharm.yaml

```diff
@@ -0,0 +1,220 @@
+  - --filesystem=xdg-videos
+  - --filesystem=xdg-download
+
+  # Need to symlink .desktop files from WineCharm/data/Prefix, if user needs them
+  - --filesystem=~/.local/share/applications
+  
+  # If the user wants to copy a wine prefix or a runner from these dirs; permissions same as io.github.fastrizwaan.WineZGUI
+  - --filesystem=~/.wine
+  - --filesystem=~/.var/app/org.winehq.Wine/data/wine:ro
+  - --filesystem=~/.var/app/com.usebottles.bottles/data/bottles:ro
+  - --filesystem=~/.var/app/com.playonlinux.PlayOnLinux4/data/PlayOnLinux/wineprefix:ro
+  - --filesystem=~/.var/app/org.phoenicis.playonlinux/.Phoenicis/containers/wineprefix:ro
+  
+  # I'm the author of WineZGUI. So WineCharm will create simple text files there, imports existing .sh scripts to WineCharm .yaml (.charm) files.
+  - --filesystem=~/.var/app/io.github.fastrizwaan.WineZGUI/data/winezgui
+
+cleanup:
+  - '*.a'
+  - '*.la'
+  - /share/doc
+  - /share/man
+  - /lib/cmake
+  - /lib/pkgconfig
+  - /share/cmake
+  - /share/pkgconfig
+  - /share/bash-completion
+
+inherit-extensions:
+  - org.freedesktop.Platform.GL32
+  - org.freedesktop.Platform.Compat.i386
+  - org.freedesktop.Platform.ffmpeg_full.i386
+  - org.freedesktop.Platform.ffmpeg-full
+  - org.freedesktop.Platform.GL32
+  - org.freedesktop.Platform.VAAPI.Intel.i386
+  - org.winehq.Wine.gecko
+  - org.winehq.Wine.mono
+
+add-extensions:
+  com.valvesoftware.Steam.Utility:
+    subdirectories: true
+    directory: utils
+    versions: stable;beta;test;master
+    add-ld-path: lib
+    merge-dirs: share/vulkan/explicit_layer.d;share/vulkan/implicit_layer.d;
+    no-autodownload: true
+    autodelete: false
+
+modules:
+  # Create 32-bit directories
+  - name: compat
+    buildsystem: simple
+    build-commands:
+      - mkdir -p /app/lib/i386-linux-gnu
+      - mkdir -p /app/lib/i386-linux-gnu/GL
+      - mkdir -p /app/lib/debug/lib/i386-linux-gnu
+      - mkdir -p /app/lib/dri/intel-vaapi-driver
+      - mkdir -p /app/lib/i386-linux-gnu/dri/intel-vaapi-driver
+      - mkdir -p /app/lib/udev/rules.d
+      - mkdir -p /app/extensions/Plugins
+      - mkdir -p /app/lib64
+      - mkdir -p /app/lib32
+      - mkdir -p /app/lib/ffmpeg
+      - mkdir -p /app/lib32/ffmpeg
+      - mkdir -p /app/utils
+
+  - name: vulkan-tools
+    buildsystem: cmake-ninja
+    config-opts:
+      - -DGLSLANG_INSTALL_DIR=/app
+      - -DVULKAN_HEADERS_INSTALL_DIR=/app
+      - -DCMAKE_BUILD_TYPE=Release
+    sources:
+      - type: git
+        url: https://github.com/KhronosGroup/Vulkan-Tools.git
+        tag: sdk-1.3.261.1
+        commit: a7da7027ca9fd0901639f02619c226da9c6036f1
+        x-checker-data:
+          type: git
+          tag-pattern: ^sdk-([\d.]+)$
+    modules:
+      - name: vulkan-tools
+        buildsystem: cmake-ninja
+        sources:
+          - type: git
+            url: https://github.com/KhronosGroup/Vulkan-Headers.git
+            tag: sdk-1.3.261.1
+            commit: 85c2334e92e215cce34e8e0ed8b2dce4700f4a50
```

Comment:
vulkan-headers is in the runtime. why does that not work?

```
flatpak run --command=find org.gnome.Sdk//48 /usr -wholename "*/vulkan.h"
/usr/include/gstreamer-1.0/gst/vulkan/vulkan.h
/usr/include/vulkan/vulkan.h
```

Reply:
bottles uses that, it is also gtk4 libadwaita app, and frontend for wine same as winecharm. So, I used that from here.

https://github.com/flathub/com.usebottles.bottles/blob/master/com.usebottles.bottles.yml#L110

Reply:
Someone else does it is not a good logic, have you tested it without? does it build and work?

Reply:
I'll rebuild and check.

Reply:
works without it.